src/PrestaShopBundle/Translation/Loader/LegacyFileLoader.php

```diff
@@ -42,56 +41,63 @@ final class LegacyFileLoader implements LoaderInterface
+    const LEGACY_TRANSLATION_FORMAT = '#\<\{(?<module>[\w-]+)\}(?<theme>[\w-]+)\>(?<domain>[\w-]+)_(?<id>[\w-]+)#';
 
     /**
-     * @var Converter the locale Converter
+     * @var LegacyFileReader
      */
-    private $localeConverter;
+    private $fileReader;
 
-    public function __construct(Converter $converter)
+    /**
+     * @var DomainNormalizer
+     */
+    private $domainNormalizer;
+
+    /**
+     * @param LegacyFileReader $fileReader
+     */
+    public function __construct(LegacyFileReader $fileReader)
     {
-        $this->localeConverter = $converter;
+        $this->fileReader = $fileReader;
+        $this->domainNormalizer = new DomainNormalizer();
     }
 
     /**
      * {@inheritdoc}
+     *
+     * Note that parameter "domain" is useless, as domain is inferred from source files
+     *
+     * @throws \PrestaShopBundle\Translation\Exception\InvalidLegacyTranslationKeyException
      */
     public function load($path, $locale, $domain = 'messages')
     {
-        // Each legacy file declare this variable to store the translations
-        $_MODULE = [];
         $catalogue = new MessageCatalogue($locale);
-        $shopLocale = $this->localeConverter->toLegacyLocale($locale);
-        $filePath = $path . "$shopLocale.php";
-
-        if (!file_exists($filePath)) {
-            throw UnsupportedLocaleException::fileNotFound($filePath, $locale);
-        }
 
-        // Load a global array $_MODULE
-        include_once $filePath;
+        $tokens = $this->fileReader->load($path, $locale);
 
-        foreach ($_MODULE as $translationKey => $translationValue) {
-            $id = $this->getId($translationKey);
-            $catalogue->set($id, $translationValue, $domain);
+        foreach ($tokens as $translationKey => $translationValue) {
+            $parsed = LegacyTranslationKey::buildFromString($translationKey);
+            $id = $parsed->getHash();
+            $catalogue->set($id, $translationValue, $this->buildDomain($parsed));
         }
 
         return $catalogue;
     }
 
     /**
-     * @param string $key the translation key
+     * Builds the domain using information in the translation key
      *
-     * @return string the translation id
+     * @param LegacyTranslationKey $translationKey
+     *
+     * @return string
      */
-    private function getId($key)
+    private function buildDomain(LegacyTranslationKey $translationKey)
     {
-        preg_match_all(self::LEGACY_TRANSLATION_FORMAT, $key, $matches);
-        if (empty($matches['id'][0])) {
-            throw LegacyFileFormattingException::fileIsInvalid();
-        }
+        $newDomain = DomainHelper::buildModuleDomainFromLegacySource(
+            $translationKey->getModule(),
+            $translationKey->getSource()
+        );
 
-        return $matches['id'][0];
+        return $this->domainNormalizer->normalize($newDomain);
```

Comment:
It could, but I'm not a fan of static methods because you cannot inject them. This class is in my view temporary because it makes no sense to use domains everywhere as "Some.Domain.Thing" and store them in the catalogue as "SomeDomainThing".

Reply:
Yeah I admit that I really don't understand on this dot removing everywhere..